internal/cmdrepoget/command.go

```diff
@@ -67,62 +67,79 @@ func newRunner(ctx context.Context, rcg *genericclioptions.ConfigFlags) *runner
 type runner struct {
 	ctx     context.Context
 	cfg     *genericclioptions.ConfigFlags
-	client  client.Client
 	Command *cobra.Command
 
 	// Flags
 	printFlags *get.PrintFlags
+
+	requestTable bool
 }
 
 func (r *runner) preRunE(cmd *cobra.Command, args []string) error {
-	const op errors.Op = command + ".preRunE"
-	client, err := porch.CreateClient(r.cfg)
-	if err != nil {
-		return errors.E(op, err)
+	outputOption := cmd.Flags().Lookup("output").Value.String()
+	if strings.Contains(outputOption, "custom-columns") || outputOption == "yaml" || strings.Contains(outputOption, "json") {
+		r.requestTable = false
+	} else {
+		r.requestTable = true
 	}
-	r.client = client
 	return nil
 }
 
 func (r *runner) runE(cmd *cobra.Command, args []string) error {
 	const op errors.Op = command + ".runE"
 
-	var objs []runtime.Object
+	// For some reason our use of k8s libraries result in error when decoding
+	// RepositoryList when we use strongly typed data. Therefore for now we
+	// use unstructured communication.
+	// The error is: `no kind "RepositoryList" is registered for the internal
+	// version of group "config.porch.kpt.dev" in scheme`. Of course there _is_
+	// no such kind since CRDs seem to have only versioned resources.
```

Comment:
Trying to remember from my `kubebuilder` days, we had to register list-kind also to the scheme. Here is an example from a kubebuilder same project: https://github.com/jetstack/kubebuilder-sample-controller/blob/master/api/v1beta1/mykind_types.go#L77

Reply:
Thank you! the List types are registered in the scheme. Problem is that the k8s implementation is looking for _unversioned_ type but those don't exist for CRDs (as far as I can tell).